src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1671,6 +1677,16 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
                 //We will call the subsequent reductions n times
                 int cartIndex = ri.CartesianIndex;
 
+                //We determine if the input paramaters are homogenious to set the final function endpoint once
+                if (cartIndex == 0)
```

Comment:
What is this check for?

Reply:
@aparajit-pratap This is so you only check the homogeneity before you get into the recursive calls of `ExecWithRISlowPath()`.  You could call these at different CartesianIndex values but you start to loose some of the performance improvement with multiple nested calls to `AreParametersHomogeneous()`.  I will double check though...  Right now I make the assumption that you only get the benefit of the predetermined `finalFunctionEndPoint` if you pass  `AreParametersHomogeneous()` with all the parameters.